extensions/yaml_support/yaml_support/_parser/lg_units_parser.py

```diff
@@ -0,0 +1,195 @@
+                                        type = self.__construct_type(elt, '')
+                                        connections.append(type)
+
+                        # transform the list of connections to a dict
+                        # each connection is represented as key-value
+                        connections_dict: Dict[str, str] = {}
+                        _iter = iter(connections)
+                        for value in _iter:
+                            key = value.split('.')[1]
+                            value = next(_iter).split('.')[1]
+
+                            if value != 'OUTPUT':
+                                connections_dict[key] = value
+                            else:
+                                connections_dict[key] = node.name
+
+                        return_info['connections_dict'] = connections_dict
+
+                    # checks for publisher & subscriber decorators
+                    publishers: List[str] = []
+                    subscribers: List[str] = []
+                    for decorator in child.decorator_list:
+                        if hasattr(decorator, 'func'):
+                            decorator_type: str = self.__construct_type(
+                                                            decorator.func,
+                                                            ''
+                                                        )
+                            decorator_type = decorator_type.split('.')[-1]
+
+                            if decorator_type in 'publisher':
+                                arg_type: str = self.__construct_type(
+                                                            decorator.args[0],
+                                                            ''
+                                                        )
+                                publishers.append(arg_type)
+
+                            elif decorator_type in 'subscriber':
+                                arg_type: str = self.__construct_type(
+                                                            decorator.args[0],
+                                                            ''
+                                                        )
+                                subscribers.append(arg_type)
+
+                    if(hasattr(child.returns, 'value')):
+                        type = self.__construct_type(child.returns, type)
+                        return_info['type'] = type
+
+                    class_model.methods[child.name] = {
+                        "args": arguments_info,
+                        "return": return_info,
+                        "publishers": publishers,
+                        "subscribers": subscribers
+                    }
+
+            assert isinstance(class_model, LabGraphUnitsModel)
+            self.__classes.append(class_model)
+            self.generic_visit(node)
+
+    def __construct_type(self, value, type):
```

Comment:
Please add return type for all functions.

Reply:
Done